src/Chapter3.hs

```diff
@@ -374,6 +384,42 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Knight = Knight 
+    { knightHealth :: Int
+    , knightAttack :: Int
+    , knightGold   :: Int
+    } deriving (Show)
+
+data Monster = Monster
+    { monsterHealth :: Int
+    , monsterAttack :: Int
+    , monsterGold   :: Int
+    } deriving (Show)
+
+mon :: Monster
+mon = Monster
+    { monsterHealth = 20
+    , monsterAttack = 2
+    , monsterGold   = 20
+    }
+
+kni :: Knight
+kni = Knight
+    { knightHealth = 25
+    , knightAttack = 2
+    , knightGold   = 15
+    }
+
+fight :: Monster -> Knight -> Int
+fight (Monster mH mA mG) (Knight kH kA kG) = 
+    if (mH - kA) > 0 then
+      if (kH - mA) > 0 then
+        -- fight (Monster (mH-kA) mA mG) (Knight (kH-mA) kA kG)
+        kG
+      else -1
+    else
+      kG + mG
```

Comment:
I feel like if you used guards here, the code will be a bit cleaner :slightly_smiling_face:

Reply:
It sure does. Thanks.